runtime/src/iree/base/internal/cpu.c

```diff
@@ -310,6 +310,74 @@ static void iree_cpu_initialize_from_platform_x86_64(uint64_t* out_fields) {
   out_fields[0] = out0;
 }
 
+#elif defined(IREE_ARCH_RISCV_64)
+#if defined(IREE_PLATFORM_ANDROID) || defined(IREE_PLATFORM_LINUX)
+
+// First 4 characters are "rv32" or "rv64"
+#define IREE_RISCV_64_ISA_SEARCH_START 4
+
+static void iree_cpu_initialize_from_platform_riscv_64(uint64_t* out_fields) {
+  char* line;
+  char isa[1024];
+  char* cpuinfo_path = "/proc/cpuinfo";
+  FILE* fp;
```

Comment:
When possible, defer local variable declaration to first-use. So below you can do `FILE* cpuinfo_file = fopen(...`.   Please use a more descriptive variable identifier than `fp`.